File: Assets/Script/character/Buff.cs
//Buff的class
public abstract class Buff
{
    public BuffKind _buffKind;
    public double _effect; //效果量：数值或者百分比
    public int _remain;

    public abstract double Count(double num);

    public Buff(BuffKind buffKind, double effect, int remain)
    {
        _buffKind = buffKind;
        _effect = effect;
        _remain = remain;
    }
}

public class AmountBuff : Buff //数值类buff
{
    public AmountBuff(BuffKind buffKind, double effect, int remain) : base(buffKind, effect, remain)
    {
    }

    public override double Count(double num)
    {
        //TODO: 计算数值
        return num;
    }
}

public class PercantageBuff : Buff //百分比buff
{
    public PercantageBuff(BuffKind buffKind, double effect, int remain) : base(buffKind, effect, remain)
    {
    }

    public override double Count(double num)
    {
        //TODO: 计算数值
        return num;
    }
}

public enum BuffKind
{
    Atk,
    Def,
    GainDamage,
    Heal,
    GainHeal
}
//TODO： 更多的buff……

File: src/Buff.java

public class Buff
{
    public BuffKind _buffKind;
    private double _amount; //效果量：数值或者百分比
    public boolean _percent; //根据此boolean值确定buff是数值buff还是百分比buff
    public int _remain;

    //buff的计算方法
    public double Count(double num)
    {
        if (_percent) //判断是数值还是百分比buff
        {
            return num * (1 + _amount / 100); //若百分比buff的_amount为15，则表示数值+15%
        }
        return num + _amount; //若数值buff的_amount为15，则表示数值+15
    }

    public Buff(BuffKind buffKind, double amount, boolean percent, int remain)
    {
        _buffKind = buffKind;
        _amount = amount;
        _percent = percent;
        _remain = remain;
    }
}

//TODO： 更多的buff……